### PR TITLE
Arguments to define victim and perpetrator compounds in create_ddi()

### LINKED ISSUE #32: Add arguments to identify victim and perpetrators in `create_ddi()`
instead of 
`create_ddi(...)`

`create_ddi(victim, perpetrator = ...)`

### PR DESCRIPTION
**Pull request by rengelke**
Closes #32

### CODE REVIEW

**Comment by Felixmil** [comment]
Several comments:
- After all, I think input types should be checked inside ddi (and the check function should be defined as private method or in utils.R.
- instead of `...` define `perpetrator` argument that can be either:
  - one compound
  - a vector of compound
  - a list of compound
- add a -for now- dummy option in `create_ddi` like `import_simulations` (boolean) and pass it to the related `DDI` method and save it as private argument. This is just to anticipate and experiment on how we'll handle future arguments for this function.

File: R/utils.R [hunk 1]
```diff
@@ -62,3 +61,24 @@ is_file_url <- function(x, error_call = caller_env()) {
...
+#' is a vector, it converts the vector to a list where each element of the vector
+#' becomes an element of the list. For all other types, it wraps the input in a
+#' list.
+#'
+#' @param x Any R object or vector.
+#' @return A list containing `x`, a list of the elements of `x` if `x` is a
+#' vector, or `x` itself if it's already a list.
+#'
+#' @keywords internal
+to_list <- function(x) {
```
Comment (Felixmil) [inline R/utils.R:77]: This is not needed, instead, use as.list() directly

```r
myvector <- c(1,2,3,4)

mylist <- list(1,2,3,4)

all.equal(as.list(myvector), as.list(mylist))

TRUE
```

File: R/ddi.R [hunk 2]
```diff
@@ -142,6 +162,36 @@ DDI <- R6::R6Class(
...
+        invalid_classes <- vapply(compounds[invalid_indices], class, character(1))
+
+        invalid_details <- paste0("[", invalid_indices, "] ", invalid_classes)
+        cli::cli_abort(c(
+          "All compounds must be of class 'Compound'. Invalid entries found at position(s):",
+          invalid_details
+        ))
+      }
+
+      # create and print status message
```
Comment (Felixmil) [inline R/ddi.R:184]: I think this is out of scope of the method. It can be moved (and simplified) to the combined method, outputing something like:
```
Combining Snapshots:
- Victim Compound: XXX
- Perpertrator(s) Compound(s):
	- YYY
	- ZZZ
```

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -13,3 +13,10 @@ test_that("is_file_local works",{
   expect_false(is_file_local(test_path("data")))
 
 })
+
+test_that("to_list works as expected", {
```
Comment (Felixmil) [inline tests/testthat/test-utils.R:17]: Instead of that, add additional tests in `test-ddi.R` that pass perpetrator as vector or list.

**Review by Felixmil — CHANGES_REQUESTED** [review]

File: R/utils.R (hunk 1 above)
Comment (rengelke) [reply R/utils.R:77]: It is still necessary when working with R6 classes:

```
mylist <- list(rifampicin)  #creates a list
mylist2 <- as.list(rifampicin) #converts R6 itself into a list
all.equal(mylist, mylist2)

> Error in unclass(current) : cannot unclass an environment


mylist <- to_list(rifampicin)
mylist2 <- to_list(rifampicin)
all.equal(mylist, mylist2)

> TRUE
```

File: R/ddi.R [hunk 4]
```diff
@@ -9,12 +15,23 @@
...
-create_ddi <- function(...) {
+create_ddi <- function(victim, perpetrator) {
+  if (missing(victim)) {
+    cli::cli_abort("At least one victim compound must be provided.")
+  }
+  if (missing(perpetrator)) {
+    cli::cli_abort("At least one perpetrator compound must be provided.")
+  }
+
+  victim <- to_list(victim)
```
Comment (Felixmil) [inline R/ddi.R:28]: Why do we convert to list ? victim should always be of length 1.

File: R/utils.R (hunk 1 above)
Comment (Felixmil) [reply R/utils.R:77]: This is true with only one element indeed.

File: R/ddi.R (hunk 4 above)
Comment (rengelke) [reply R/ddi.R:28]: Yes, we don't really need it.

**Review by Felixmil — APPROVED** [review]
